packages/plugin-core/src/components/views/PreviewPanel.ts

```diff
@@ -0,0 +1,279 @@
+import {
+  assertUnreachable,
+  DendronEditorViewKey,
+  DMessageEnum,
+  getWebEditorViewEntry,
+  NoteProps,
+  NoteUtils,
+  NoteViewMessage,
+  NoteViewMessageEnum,
+  OnDidChangeActiveTextEditorMsg,
+} from "@dendronhq/common-all";
+import _ from "lodash";
+import * as vscode from "vscode";
+import { IDendronExtension } from "../../dendronExtensionInterface";
+import { Logger } from "../../logger";
+import { INoteSyncService } from "../../services/NoteSyncService";
+import { sentryReportingCallback } from "../../utils/analytics";
+import { WebViewUtils } from "../../views/utils";
+import { VSCodeUtils } from "../../vsCodeUtils";
+import { WSUtilsV2 } from "../../WSUtilsV2";
+import { IPreviewLinkHandler } from "./PreviewLinkHandler";
+import { PreviewProxy } from "./PreviewProxy";
+
+/**
+ * This is the default implementation of PreviewProxy. It contains a singleton
+ * of a vscode webviewPanel that renders the note preview. Furthermore, it will
+ * automatically handle event subscriptions to know when to update the preview,
+ * as well as properly dispose of the resources when the preview has been
+ * closed.
+ */
+export class PreviewPanel implements PreviewProxy, vscode.Disposable {
+  private _ext: IDendronExtension;
+  private _panel: vscode.WebviewPanel | undefined;
+  private _noteSyncService: INoteSyncService;
+  private _onDidChangeActiveTextEditor: vscode.Disposable | undefined =
+    undefined;
+  private _onNoteChanged: vscode.Disposable | undefined = undefined;
+  private _linkHandler: IPreviewLinkHandler;
+
+  /**
+   *
+   * @param param0 extension - IDendronExtension implementation. linkHandler -
+   * Implementation to handle preview link clicked events
+   */
+  constructor({
+    extension,
+    linkHandler,
+  }: {
+    extension: IDendronExtension;
+    linkHandler: IPreviewLinkHandler;
+  }) {
+    this._ext = extension;
+    this._linkHandler = linkHandler;
+    this._noteSyncService = this._ext.noteSyncService;
+  }
+
+  /**
+   * Show the preview.
+   * @param note - if specified, this will override the preview contents with
+   * the contents specified in this parameter. Otherwise, the contents of the
+   * preview will follow default behavior (it will show the currently in-focus
+   * Dendron note).
+   */
+  show(note?: NoteProps): void {
+    if (this._panel) {
```

Comment:
Nit, but you can reuse your methods you defined. `if (this.isOpen() && !this.isVisible())`

Reply:
made some slight modifications here - but one Typescript thing - if it's not explicit that `this._panel` is not undefined via the if condition, then its usage on line 67 will result in a 'object possibly undefined' warning.